src/main/java/org/alliancegenome/curation_api/services/base/BaseOntologyTermService.java

```diff
@@ -227,24 +234,28 @@ private void handleSynonyms(OntologyTerm dbTerm, OntologyTerm incomingTerm) {
 		}
 		List<String> newSynonymNames = newSynonyms.stream().map(Synonym::getName).toList();
 
-		for (Synonym syn: newSynonyms) {
+		for (Synonym syn : newSynonyms) {
+			SearchResponse<Synonym> response = synonymDAO.findByField("name", syn.getName());
 			if (!currentSynonymNames.contains(syn.getName())) {
-				SearchResponse<Synonym> response = synonymDAO.findByField("name", syn.getName());
 				Synonym synonym;
 				if (response == null) {
 					synonym = synonymDAO.persist(syn);
 				} else {
 					synonym = response.getSingleResult();
 				}
 				dbTerm.getSynonyms().add(synonym);
+			} else {
+				Synonym dbSynonym = response.getSingleResult();
+				if (!sameSynonym(dbSynonym, syn)) {
+					updateSynonym(dbSynonym, syn);
+				}
```

Comment:
@adamgibs the other option here is to just always update.

Reply:
So remove  the equality check and just update in the else statement always?

Reply:
Ya always update is easier than trying to compute the diff. If this gets to resource intensive we can always revisit it.